pydantic_graph/pydantic_graph/graph.py

```diff
@@ -126,14 +126,15 @@ def __init__(
 
         self._validate_edges()
 
-    async def run(
+    def run(
         self: Graph[StateT, DepsT, T],
         start_node: BaseNode[StateT, DepsT, T],
         *,
         state: StateT = None,
         deps: DepsT = None,
         infer_name: bool = True,
-    ) -> tuple[T, list[HistoryStep[StateT, T]]]:
+        span: LogfireSpan | None = None,
+    ) -> GraphRun[StateT, DepsT, T]:
         """Run the graph from a starting node until it ends.
```

Comment:
I think we could do with adding a prominent message here explaining that this can behave like a coroutine or a context manager etc.